### PR TITLE
translate: basic/prefix-sum.md

### PR DESCRIPTION
**Pull request by CamberLoid**
Please **Reviewers** take care of the following aspects:

1. Are there any typos?
2. Please read the content of the Pull Request as if you were a beginner and talk about your thoughts after reading the text no matter whether you are familiar with the content or not.
3. Please evaluate the content of the Pull Request if you are familiar with related concepts.
4. Please **try your best** to keep an eye on it until it is merged or closed.

### CODE REVIEW

File: docs/basic/prefix-sum.md [hunk 1]
```diff
@@ -1,198 +1,372 @@
...
+
+    FJ is pumping milk between K
+    pairs of stalls $K(1 \le K \le 100,000)$. For the ith such pair, you are told two stalls si and ti, endpoints of a path along which milk is being pumped at a unit rate. FJ is concerned that some stalls might end up overwhelmed with all the milk being pumped through them, since a stall can serve as a waypoint along many of the K paths along which milk is being pumped. Please help him determine the maximum amount of milk being pumped through any stall. If milk is being pumped along a path from si to ti, then it counts as being pumped through the endpoint stalls si and ti, as well as through every stall along the path between them. 
+
+??? note "解题思路"
+    Because it is needed to count how many times has been every node accessed, by using the method of adjacent difference on tree to add $1$ on every node on every path, we can quickly get the number of accessing of nodes. Here uses binary lifting to calculate LCA. In the end by using DFS to traverse the whole tree, we can find the answer by calculate the sum of adjacent difference array when backtracking.
+
+??? note "参考代码"
```
Comment (ksyx) [inline docs/basic/prefix-sum.md:243]: 标题

File: docs/basic/prefix-sum.md [hunk 2]
```diff
@@ -1,198 +1,372 @@
...
+    Because of the addition of $sum_{i - 1,j}$ and $sum_{i,j - 1}$ is duplicated with $sum_{i - 1,j - 1}$, here subtracted it.
+    
+    The second problem is how to apply, e.g., calculate the sum of sub-matrixes $(x1,y1) - (x2,y2)$.
+    
+    Then we can easily conclude that $sum_{x2,y2} - sum_{x1 - 1,y2} - sum_{x2,y1 - 1} + sum_{x1 - 1,y1 - 1}$ based on similar idea.
+
+#### Example
+
+???+note "[Luogu P1387](https://www.luogu.com.cn/problem/P1387)"
+    Given a $n*m$ matrix with only $0$ and $1$. Your task is to find the largest square without $0$ in the matrix and print the length of its edge.
```
[the remark this answers is not in the document]
Comment (CamberLoid) [reply docs/basic/prefix-sum.md:96]: 啊，咱差点想用\bullet

File: docs/basic/prefix-sum.md [hunk 3]
```diff
@@ -1,198 +1,372 @@
...
+    }
+    ```
 
-int N, A[10000], B[10000];
-int main() {
-  cin >> N;
-  for (int i = 0; i < N; i++) {
-    cin >> A[i];
-  }
+### High-Dimensional Sum over Subsets Dynamic Programming
```
[the remark this answers is not in the document]
Comment (CamberLoid) [reply docs/basic/prefix-sum.md:136]: Ref: https://codeforces.com/blog/entry/45223

> Sum over Subsets Dynamic Programming

[the remark this answers is not in the document]
Comment (CamberLoid) [reply docs/basic/prefix-sum.md:135]: ping @ksyx

[the remark this answers is not in the document]
Comment (CamberLoid) [reply docs/basic/prefix-sum.md:135]: #Ignored

**Pull request merged**